File: Cpp/566_Reshape_the_Matrix.cpp
class Solution {
public:
    vector<vector<int>> matrixReshape(vector<vector<int>>& nums, int r, int c) {
        if (nums.size() * nums[0].size() != r*c) return nums;
        
        vector<vector<int>> result(r, vector<int>(c));
        int i=0;
        for(auto& row : nums){
            for(auto& col : row){
                result[i/c][i%c] = col;
                ++i;
            }
        }
        
        return result;
    }
};

File: Kotlin/566_Reshape_the_Matrix.kt
class Solution {
    fun matrixReshape(mat: Array<IntArray>, r: Int, c: Int): Array<IntArray> {
        if (mat.size * mat[0].size != r * c) return mat
        
        val result = Array(r) { IntArray(c) }
        var i = 0
        mat.forEach{ row ->
            row.forEach{ col ->
                result[i / c][i % c] = col
                ++i
            }
        }
        
        return result
    }
}